From 88463f033dacf0a605c7cdb671a359f5d6caac9f Mon Sep 17 00:00:00 2001
From: Pankaj Maharana <pankaj.maharana@lge.com>
Date: Sun, 1 Dec 2019 20:41:59 +0530
Subject: [PATCH] bcm2835-v4l2-codec: fix vchiq mmal renable

:Release Notes:
Enable vchiq mmal component on stream start after stream stop

:Detailed Notes:
Enable vchiq mmal component on stream start after stream stop
When we stop streaming for flush from gst vrl2 decoder element
then vchiq mmal component is disabled.
But it does not start back when we start the stream back from
gst v4l2 decoder element because the ctx->component_enabled flag
is not set to false while disabling the component. So setting
the flag to false while disabling vchiq mmal component
Also fixes an error in vidioc_s_selection that is using height
instead of crop_height. Might be a typo it is effecting in
dynamic resolution change of the stream.

:Testing Performed:
Tested the 1080p video plays with 22fps in raspberrypi4

:QA Notes:

:Issues Addressed:
[PLAT-94493] Streaming issues in bcm2835-v4l2-codec driver
---
 drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
index 66edec1..1986118 100644
--- a/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
+++ b/drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c
@@ -1342,7 +1342,7 @@ static int vidioc_s_selection(struct file *file, void *priv,
 			s->r.left = 0;
 			s->r.top = 0;
 			s->r.width = min(s->r.width, q_data->crop_width);
-			s->r.height = min(s->r.height, q_data->height);
+			s->r.height = min(s->r.height, q_data->crop_height);
 			q_data->crop_width = s->r.width;
 			q_data->crop_height = s->r.height;
 			q_data->selection_set = true;
@@ -2201,6 +2201,7 @@ static void bcm2835_codec_stop_streaming(struct vb2_queue *q)
 		if (ret)
 			v4l2_err(&ctx->dev->v4l2_dev, "%s: Failed enabling component, ret %d\n",
 				 __func__, ret);
+		ctx->component_enabled = false;
 	}
 
 	if (V4L2_TYPE_IS_OUTPUT(q->type))
-- 
2.7.4

